### PR TITLE
Added esm bundle to output

### PR DESCRIPTION
**Pull request by shkreios**
## Current state
The library only supplies a `cjs` & `d.ts` output.

## Why change
Even though bundlers, e.g. vite, have support for transforming `cjs` into `esm` it's nice when the transformation happens at library level. 

## Result
 This pr adds a `esm` bundle to the output and updates `package.json` fields.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,9 @@
   "name": "react-use-event-hook",
   "version": "0.9.2",
   "description": "Same as React's `useCallback`, but returns a stable reference.",
-  "main": "dist/useEvent.js",
+  "main": "dist/index.js",
+  "types": "dist/useEvent.d.ts",
+  "module": "dist/useevent.mjs",
```
Comment (scottrippey) [inline package.json:7]: In general, this sounds like a great addition.  Supporting ESM is a good idea.

However, looking at the compiled `tsup` output, I'm disappointed by the extra complexity.  I'm a fan of simply using `tsc`, and its ESM output is much cleaner and easier to read or debug.  

I would love if you could update this PR to simply use `tsc`, instead of introducing new dependencies.

File: package.json [hunk 2]
```diff
@@ -2,7 +2,9 @@
   "name": "react-use-event-hook",
   "version": "0.9.2",
   "description": "Same as React's `useCallback`, but returns a stable reference.",
-  "main": "dist/useEvent.js",
+  "main": "dist/index.js",
```
Comment (scottrippey) [inline package.json:5]: Outputting to 2 sibling files (eg. `useEvent.js, useEvent.mjs`) does seem have some problems with some build tools.

There are some solutions listed here: https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html

TL;DR; I think it's best to output to 2 separate folders, eg. `dist/cjs/useEvent.js` and `dist/mjs/useEvent.mjs`

**Review by scottrippey — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (scottrippey) [reply package.json:7]: Since I was experimenting with this, I created #10

File: package.json (hunk 2 above)
Comment (scottrippey) [reply package.json:5]: See #10
